### PR TITLE
Add auto-publishing docs with RTD integration

### LINKED ISSUE #93: Auto-deploy python documentation
Would be nice if we had documentation automatically deployed onto Read the Docs. More specifically, the API docs on database, commands, and maybe a short little tutorial on how everything works together, and how to build a plugin.

Might need some way to access through ubclaunchpad (since this repo is under the organization).

### PR DESCRIPTION
**Pull request by cheukyin699**
# Pull Request

## Description

Adds readthedocs python sphinx integration. Rewrite/Reformat some docs to better fit the internet, and for better reading.

In the future, tutorials could be added, or at least usage guides and quick-starting guides on using the plugin "system" (whenever we finish our first `command` plugin entirely). Documentation on when to use certain commands could also work. We go with DDD (Documentation Driven Development).

> Beautiful docs = beautiful code
> -- Elbert Einstein

## Testing

Check out the latest version of the docs (this branch) at https://rocket20.readthedocs.io/en/cheukyin699--93-publish-docs/

## Ticket(s)

Closes #93.

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -18,6 +18,9 @@ pytest-cov = "*"
 pydocstyle = "*"
 ipython = "*"
 awscli = "*"
+sphinx = "*"
+sphinx-rtd-theme = "*"
+m2r = "*"
```
Comment (rwblickhan) [inline Pipfile]: What is `m2r`?

Comment (cheukyin699) [reply Pipfile]: markdown to rst, used to convert markdown to reStructuredText, though some things don't convert 100%

File: docs/Scripts.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+to `.git/hooks/post-merge`.
+
+After installation, every time you try to make a commit, all the tests will be
+run automatically to ensure compliance. Every time you perform a `pull` or
+`merge` or `rebase`, `pipenv` will try to sync all packages and dependencies.
+
+## Makefile for Documentation
+
+```sh
+make clean html
```
[the remark this answers is not in the document]
Comment (cheukyin699) [reply docs/Scripts.md]: It's built into sphinx. It runs clean (which deletes any remnants of the `_build/` directory), and then builds `html`. I've found lots of caching issues if you just run `make html`; lots of things don't get updated correctly.

**Comment by bobheadxi** [comment]
Just gotta say this is **super** nice, I love it, awesome work @cheukyin699 !

File: README.md [hunk 3]
```diff
@@ -3,6 +3,7 @@
 [![Build Status](https://travis-ci.org/ubclaunchpad/rocket2.0.svg?branch=master)](https://travis-ci.org/ubclaunchpad/rocket2.0)
 [![codecov](https://codecov.io/gh/ubclaunchpad/rocket2.0/branch/master/graph/badge.svg)](https://codecov.io/gh/ubclaunchpad/rocket2.0)
 [![Deployed with Inertia](https://img.shields.io/badge/deploying%20with-inertia-blue.svg)](https://github.com/ubclaunchpad/inertia)
+[![Documentation Status](https://readthedocs.org/projects/rocket20/badge/?version=latest)](https://rocket20.readthedocs.io/en/latest/?badge=latest)
```
Comment (rwblickhan) [inline README.md]: 😃

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# Minimal makefile for Sphinx documentation
+#
```
Comment (joy8zhang) [inline Makefile]: Is this empty comment line needed?

Comment (cheukyin699) [reply Makefile]: i kept it in there for... fun

**Pull request merged**